src/steps/generic.rs

```diff
@@ -151,3 +152,43 @@ pub fn run_remote_topgrade(run_type: RunType, hostname: &str) -> Result<(), Erro
         ])
         .check_run()
 }
+
+pub fn run_sdkman(base_dirs: &BaseDirs, cleanup: bool, run_type: RunType) -> Result<(), Error> {
+    let shell_path = env::var("SHELL")
+        .map(PathBuf::from)
+        .map_err(|_| Error::from(ErrorKind::SkipStep))
+        .and_then(PathExt::require)
+        .or_else(|_| utils::require("zsh"))
+        .or_else(|_| utils::require("bash"))
+        .or_else(|_| utils::require("sh"))
+        .map(|p| format!("{}", &p.display()))?;
+
+    let sdkman_init_path = env::var("SDKMAN_DIR")
+        .map(PathBuf::from)
+        .unwrap_or_else(|_| base_dirs.home_dir().join(".sdkman"))
+        .join("bin")
+        .join("sdkman-init.sh")
+        .require()
+        .map(|p| format!("{}", &p.display()))?;
+
+    print_separator("SDKMAN!");
+
+    let cmd_update = format!(
```

Comment:
Do we really have to execute the shell ourselves? I use fish as my shell so `source` will probably fail on my computer. If there's no way other then to execute the shell ourselves then I suggest that we always execute bash, as that's the shell SDKMAN is written in.

Reply:
I think that if Topgrade is compiled in Cygwin then Rust will set the `unix` configuration instead of `windows`. That's the assumption that I wrote the other steps with.

Reply:
Well, SDKMAN has no binary, but is invoked by a bash function sourced from sdkman-init.sh, so I'm out of ideas when it comes to make it run without executing shell.
Any suggestions?
Otherwise I will change the code to check for bash and execute every command in it's statement.

Reply:
I see. If that's the case then I suggest that we always use bash instead of `SHELL`, because `SHELL` might point to a shell that's not bash compatible, such as fish.